src/api/new/managers/AuthManager.ts

```diff
@@ -0,0 +1,223 @@
+    console.log('Response Body:', body);
+
+    if (response.status !== 200) {
+      console.error(
+        'Authentication request returned status:',
+        response.status,
+        ' ',
+        response.statusText,
+        '\nBody:',
+        body
+      );
+      return;
+    }
+
+    // Log the response body
+    console.log('Response Body:', body);
+
+    const accessToken: AccessToken = {
+      access_token: body.access_token,
+      token_type: body.token_type,
+      refresh_token: body.refresh_token,
+      expires_in: body.expires_in,
+      expires: Date.now() + body.expires_in * 1000,
+    };
+
+    localStorage.setItem('reactify:access_token', JSON.stringify(accessToken));
+
+    setTimeout(this.refreshToken, 1_800_000); // 30 minutes
+  }
+
+  public async refreshToken(): Promise<void> {
+    const accessToken = this.getAccessToken();
+
+    if (!accessToken) {
+      return Promise.reject('No access token found in localStorage');
+    }
+
+    const refreshToken = accessToken.refresh_token;
+    const url = env.AUTH_REQUEST_TOKEN_URL;
+
+    const params = {
+      grant_type: 'refresh_token',
+      refresh_token: refreshToken,
+      client_id: this.spotify.getClientId(),
+    };
+
+    const payload = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: new URLSearchParams(params),
+    };
+
+    const response = await fetch(url, payload);
+
+    if (response.status !== 200) {
+      return Promise.reject(`Trying to refresh token returned status ${response.status}.`);
+    }
+
+    const body = (await response.json()) as TokenRefreshResponse;
+
+    const newAccessToken: AccessToken = {
+      access_token: body.access_token,
+      token_type: body.token_type,
+      refresh_token: body.refresh_token,
+      expires_in: body.expires_in,
+      expires: Date.now() + body.expires_in * 1000,
+    };
+
+    localStorage.setItem('reactify:access_token', JSON.stringify(newAccessToken));
+
+    setTimeout(this.refreshToken, 1_800_000); // 30 minutes
+  }
+
+  public getAccessToken(): AccessToken | null {
+    const lsValue = localStorage.getItem('reactify:access_token');
+
+    if (!lsValue) return null;
+    const token = JSON.parse(lsValue) as AccessToken;
+
+    if (!(token.access_token && token.token_type && token.expires_in && token.refresh_token && token.expires)) {
+      return null;
+    }
+
+    return token;
+  }
+
+  public isLoggedIn(): boolean {
+    const accessToken = this.getAccessToken();
+
+    return (
+      !!accessToken &&
+      !!accessToken.access_token &&
+      !!accessToken.token_type &&
+      !!accessToken.expires_in &&
+      !!accessToken.refresh_token &&
+      !!accessToken.expires
+    );
+  }
+
+  private generateState(): string {
+    return window.crypto.randomUUID().toString();
+  }
+
+  private validateState(state: string): boolean {
+    return state === localStorage.getItem('reactify:auth_state');
+  }
+
```

Comment:
Should be renamed as this method is also called after successful authentication at which point it's no longer "aborting" but "cleaning" the data from local storage.